src/components/timeline/LocalTrack.js

```diff
@@ -0,0 +1,180 @@
+      // This is needed to allow the context menu to know what was right clicked without
+      // actually changing the current selection.
+      if (trackReference) {
+        changeRightClickedTrack(trackReference);
+      }
+    }
+  };
+
+  _onLineClick = () => {
+    const { threadIndex, changeSelectedThread } = this.props;
+    if (threadIndex !== null) {
+      changeSelectedThread(threadIndex);
+    }
+  };
+
+  renderTrack() {
+    const { localTrack } = this.props;
+    switch (localTrack.type) {
+      case 'thread':
+        return <TrackThread threadIndex={localTrack.threadIndex} />;
+      case 'network':
+      case 'memory':
+        // TODO: Add support for these track types.
+        return <div />;
+      default:
+        console.error('Unhandled localTrack type', (localTrack: empty));
+        return null;
+    }
+  }
+
+  render() {
+    const { isSelected, isHidden, titleText, trackName, style } = this.props;
+
+    if (isHidden) {
+      // If this global track is hidden, render out a stub element so that the
+      // Reorderable Component still works across all the tracks.
+      return <li className="timelineTrackHidden" />;
+    }
+
+    return (
+      <li className="timelineTrack timelineTrackLocal" style={style}>
+        {/* This next div is used to mirror the structure of the TimelineGlobalTrack */}
+        <div
+          className={classNames('timelineTrackRow timelineTrackLocalRow', {
+            selected: isSelected,
+          })}
+          onClick={this._onLineClick}
+        >
+          <ContextMenuTrigger
+            id={'TimelineTrackContextMenu'}
+            renderTag="div"
+            attributes={{
+              title: titleText,
+              className:
+                'timelineTrackLabel timelineTrackLocalLabel timelineTrackLocalGrippy',
+              onMouseDown: this._onLabelMouseDown,
+            }}
+          >
+            <h1 className="timelineTrackName">{trackName}</h1>
+          </ContextMenuTrigger>
+          <div className="timelineTrackTrack">{this.renderTrack()}</div>
+        </div>
+      </li>
+    );
+  }
+}
+
+const options: ExplicitConnectOptions<OwnProps, StateProps, DispatchProps> = {
+  mapStateToProps: (state, { pid, localTrack, trackIndex }) => {
+    // These get assigned based on the track type.
+    let threadIndex = null;
+    let isSelected = false;
+    let titleText = null;
+    let trackName;
+
+    // Run different selectors based on the track type.
+    switch (localTrack.type) {
+      case 'thread':
+        {
+          // Look up the thread information for the process if it exists.
+          threadIndex = localTrack.threadIndex;
+          const selectors = selectorsForThread(threadIndex);
+          isSelected = threadIndex === getSelectedThreadIndex(state);
+          trackName = selectors.getFriendlyThreadName(state);
+          titleText = selectors.getThreadProcessDetails(state);
+        }
+        break;
```

Comment:
nit: if you put the `break` inside the bracket above, the indentation will be a bit better